babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -991,30 +991,38 @@ SqlNode CurrentDateFunction() :
 
 SqlNode DateTimeTerm() :
 {
-    final SqlNode e;
-    SqlIdentifier timeZoneValue;
+    final SqlNode dateTimePrimary;
+    final SqlNode displacement;
 }
 {
     (
-        e = DateTimeLiteral()
+        dateTimePrimary = DateTimeLiteral()
     |
-        e = SimpleIdentifier()
+        dateTimePrimary = SimpleIdentifier()
     |
-        e = DateFunctionCall()
+        dateTimePrimary = DateFunctionCall()
     )
     (
         <AT>
         (
             <LOCAL>
             {
-                return new SqlDateTimeAtLocal(getPos(), e);
+                return new SqlDateTimeAtLocal(getPos(), dateTimePrimary);
             }
         |
-            <TIME> <ZONE>
-            {
-                timeZoneValue = SimpleIdentifier();
-                return new SqlDateTimeAtTimeZone(getPos(), e, timeZoneValue);
-            }
+            (
+                [<TIME> <ZONE>]
+                (
+                    displacement = SimpleIdentifier()
+                |
+                    displacement = IntervalLiteral()
+                |
+                    displacement = NumericLiteral()
+                )
+                {
+                    return new SqlDateTimeAtTimeZone(getPos(), dateTimePrimary, displacement);
+                }
+            )
```

Comment:
Seems like the parentheses are not useful here, so you could just remove them and un-indent this section one level.

Reply:
Update in https://github.com/googleinterns/calcite/pull/91/commits/5369c9eeb42f4f4ffe15a004609aebfcfcabf0c9

Reply:
In fact, I think the outer parentheses (starting on line 998 and ending on line 1025) can probably be removed as well?

Reply:
How about the outer parens mentioned in my comment above? Can we remove those too?

Reply:
I did not happen to see Drew's comments on the outer parentheses. It is now removed in https://github.com/googleinterns/calcite/pull/91/commits/6b6aed1906fcb75730c7d0c4af28022380bb2cf6